provider/internal/builtin/service_client.go

```diff
@@ -47,16 +47,38 @@ func (p *builtinServiceClient) Evaluate(ctx context.Context, cap string, conditi
 	if err != nil {
 		return provider.ProviderEvaluateResponse{}, fmt.Errorf("unable to get query info: %v", err)
 	}
+	log := p.log.WithValues("ruleID", cond.ProviderContext.RuleID)
+	log.V(5).Info("builtin condition context", "condition", cond, "provider context", cond.ProviderContext)
 	response := provider.ProviderEvaluateResponse{Matched: false}
 	switch cap {
 	case "file":
 		c := cond.File
 		if c.Pattern == "" {
 			return response, fmt.Errorf("could not parse provided file pattern as string: %v", conditionInfo)
 		}
-		matchingFiles, err := findFilesMatchingPattern(p.config.Location, c.Pattern)
-		if err != nil {
-			return response, fmt.Errorf("unable to find files using pattern `%s`: %v", c.Pattern, err)
+		matchingFiles := []string{}
+		if ok, paths := cond.ProviderContext.GetScopedFilepaths(); ok {
+			regex, _ := regexp.Compile(c.Pattern)
+			for _, path := range paths {
+				matched := false
+				if regex != nil {
+					matched = regex.MatchString(path)
+				} else {
+					// TODO(fabianvf): is a fileglob style pattern sufficient or do we need regexes?
+					matched, err = filepath.Match(c.Pattern, path)
+					if err != nil {
+						continue
+					}
+				}
+				if matched {
+					matchingFiles = append(matchingFiles, path)
+				}
+			}
+		} else {
```

Comment:
The problem here is that if we have a rule with filepaths and scopes are also added, its filepaths are ignored? That would essentially change the rule. Sorry if I'm not understanding this correctly.

Reply:
I tried to answer this question above as well as that question, but you are basically on point. 

say a rule is searching for web.xml (hardcode or found) but when we set the scope of the analysis, we only want pom.xml. In this case I **think** the right thing to do, is only search the filepaths that is the intersection of scope of the analysis and the filepaths from the rule. 

Basically I imagine the scope, is as if you copied the included paths to $PLACE and only ran analysis on $PLACE. 

Does that help answer the question?

Reply:
Makes sense, thanks!